--- tools/cmake/include/BoostInstall.cmake.old	2026-01-08 11:26:12.013228000 +0100
+++ tools/cmake/include/BoostInstall.cmake	2026-01-08 13:59:42.824649975 +0100
@@ -337,6 +337,22 @@
     string(APPEND CONFIG_INSTALL_DIR "-static")
   endif()
 
+  if(NOT CMAKE_VERSION VERSION_LESS 3.28)
+    get_target_property(INTERFACE_CXX_MODULE_SETS ${LIB} INTERFACE_CXX_MODULE_SETS)
+    if(INTERFACE_CXX_MODULE_SETS)
+      boost_message(DEBUG "boost_install_target: '${__TARGET}' has INTERFACE_CXX_MODULE_SETS=${INTERFACE_CXX_MODULE_SETS}")
+      set(__INSTALL_CXX_MODULES FILE_SET ${INTERFACE_CXX_MODULE_SETS} DESTINATION ${CMAKE_INSTALL_DATADIR})
+    endif()
+  endif()
+
+  if(NOT CMAKE_VERSION VERSION_LESS 3.23)
+    get_target_property(INTERFACE_HEADER_SETS ${LIB} INTERFACE_HEADER_SETS)
+    if(INTERFACE_HEADER_SETS)
+      boost_message(DEBUG "boost_install_target: '${__TARGET}' has INTERFACE_HEADER_SETS=${INTERFACE_HEADER_SETS}")
+      set(__INSTALL_HEADER_SETS FILE_SET ${INTERFACE_HEADER_SETS})
+    endif()
+  endif()
+
   install(TARGETS ${LIB} EXPORT ${LIB}-targets
     # explicit destination specification required for 3.13, 3.14 no longer needs it
     RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
@@ -344,8 +360,18 @@
     ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
     PRIVATE_HEADER DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}"
     PUBLIC_HEADER DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}"
+    # explicit needed if used starting with cmake v3.28
+    # XXX FILE_SET CXX_MODULES DESTINATION "${CMAKE_INSTALL_DATADIR}"
+    ${__INSTALL_CXX_MODULES}
+    # Any module files from C++ modules from PUBLIC sources in a file set of type CXX_MODULES will be installed to the given DESTINATION.
+    # FIXME: why does this silently not work here? CK
+    # XXX CXX_MODULES_BMI DESTINATION ${CONFIG_INSTALL_DIR}/bmi-${CMAKE_CXX_COMPILER_ID}_$<CONFIG>
+    # explicit needed if used starting with cmake v3.23
+    # XXX FILE_SET HEADERS
+    ${__INSTALL_HEADER_SETS}
   )
 
+  # TODO(CK): what is this for?
   export(TARGETS ${LIB} NAMESPACE Boost:: FILE export/${LIB}-targets.cmake)
 
   if(MSVC)
@@ -362,7 +388,9 @@
     __boost_install_update_sources(${LIB} ${__EXTRA_DIRECTORY} ${__EXTRA_INSTALL_DIRECTORY})
   endif()
 
-  install(EXPORT ${LIB}-targets DESTINATION "${CONFIG_INSTALL_DIR}" NAMESPACE Boost:: FILE ${LIB}-targets.cmake)
+  install(EXPORT ${LIB}-targets DESTINATION "${CONFIG_INSTALL_DIR}" NAMESPACE Boost:: FILE ${LIB}-targets.cmake
+    # XXX CXX_MODULES_DIRECTORY .
+  )
 
   set_target_properties(${LIB} PROPERTIES _boost_is_installed ON)
 
--- tools/cmake/include/BoostRoot.cmake.old	2026-01-08 11:31:49.525898365 +0100
+++ tools/cmake/include/BoostRoot.cmake	2026-01-08 11:31:04.000000000 +0100
@@ -123,7 +123,9 @@
     set(BOOST_SKIP_INSTALL_RULES ON)
   endif()
 
-  set(BUILD_TESTING OFF)
+  if(NOT BOOST_BUILD_TESTING)
+    set(BUILD_TESTING OFF)
+  endif()
 
 endif()
 
@@ -303,6 +305,9 @@
 
 endmacro()
 
+# C++20 modules
+option(BOOST_USE_MODULES "Build Boost as a collection of C++20 modules (unstable)" OFF)
+
 #
 
 file(GLOB __boost_libraries RELATIVE "${BOOST_SUPERPROJECT_SOURCE_DIR}/libs" "${BOOST_SUPERPROJECT_SOURCE_DIR}/libs/*/CMakeLists.txt" "${BOOST_SUPERPROJECT_SOURCE_DIR}/libs/numeric/*/CMakeLists.txt")
--- libs/serialization/CMakeLists.txt.old	2025-12-10 17:42:32.995666000 +0100
+++ libs/serialization/CMakeLists.txt	2026-01-08 18:20:48.675263056 +0100
@@ -6,6 +6,8 @@
 
 project(boost_serialization VERSION "${BOOST_SUPERPROJECT_VERSION}" LANGUAGES CXX)
 
+set(CMAKE_UNITY_BUILD OFF)
+
 add_library(boost_serialization
   src/archive_exception.cpp
   src/basic_archive.cpp
