--- libs/lexical_cast/include/boost/lexical_cast/detail/lcast_precision.hpp.old	2026-08-19 07:56:48.000000000 +0200
+++ libs/lexical_cast/include/boost/lexical_cast/detail/lcast_precision.hpp	2026-08-19 08:19:30.055994324 +0200
@@ -10,6 +10,8 @@
 
 #include <boost/lexical_cast/detail/config.hpp>
 
+#include <climits> // ULONG_MAX
+
 #if !defined(BOOST_USE_MODULES) || defined(BOOST_LEXICAL_CAST_INTERFACE_UNIT)
 
 #ifndef BOOST_LEXICAL_CAST_INTERFACE_UNIT
@@ -18,9 +20,8 @@
 #   pragma once
 #endif
 
-#include <climits>
 #include <ios>
-#include <limits>
+#include <boost/limits.hpp>
 
 #endif  // #ifndef BOOST_LEXICAL_CAST_INTERFACE_UNIT
 
--- libs/lexical_cast/include/boost/lexical_cast/detail/lcast_unsigned_converters.hpp.old	2026-08-19 07:24:00.637556000 +0200
+++ libs/lexical_cast/include/boost/lexical_cast/detail/lcast_unsigned_converters.hpp	2026-08-19 08:17:06.236665016 +0200
@@ -20,6 +20,8 @@
 
 #include <boost/lexical_cast/detail/config.hpp>
 
+#include <climits> // CHAR_MAX
+
 #if !defined(BOOST_USE_MODULES) || defined(BOOST_LEXICAL_CAST_INTERFACE_UNIT)
 
 #ifndef BOOST_LEXICAL_CAST_INTERFACE_UNIT
@@ -28,7 +30,6 @@
 #   pragma once
 #endif
 
-#include <climits>
 #include <cstddef>
 #include <string>
 #include <cstring>
