--- libs/stacktrace/include/boost/stacktrace/detail/addr2line_impls.hpp.old	2026-08-19 11:32:34.481485000 +0200
+++ libs/stacktrace/include/boost/stacktrace/detail/addr2line_impls.hpp	2026-08-19 11:39:49.341239746 +0200
@@ -12,11 +12,12 @@
 #   pragma once
 #endif
 
+#include <cstring>  // std::strchr
+#include <cstdio>   // fclose, fdopen, dup2
+
 #if !defined(BOOST_STACKTRACE_INTERFACE_UNIT)
 #include <boost/core/demangle.hpp>
 
-#include <cstdio>
-#include <cstring>
 
 #include <sys/types.h>
 #include <sys/wait.h>
