--- tools/cmake/include/BoostInstall.cmake.old	2026-08-20 18:18:48.000000000 +0200
+++ tools/cmake/include/BoostInstall.cmake	2026-08-21 10:38:47.326224762 +0200
@@ -337,6 +337,25 @@
     string(APPEND CONFIG_INSTALL_DIR "-static")
   endif()
 
+  set(__INSTALL_CXX_MODULES)
+  if(NOT CMAKE_VERSION VERSION_LESS 3.28)
+    get_target_property(INTERFACE_CXX_MODULE_SETS ${LIB} INTERFACE_CXX_MODULE_SETS)
+    if(INTERFACE_CXX_MODULE_SETS)
+      boost_message(DEBUG "boost_install_target: '${__TARGET}' has INTERFACE_CXX_MODULE_SETS=${INTERFACE_CXX_MODULE_SETS}")
+      set(extrainstalldir "${CMAKE_INSTALL_DATADIR}/boost-${__VERSION}/${LIB}")
+      set(__INSTALL_CXX_MODULES FILE_SET ${INTERFACE_CXX_MODULE_SETS} DESTINATION ${extrainstalldir})
+    endif()
+  endif()
+
+  set(__INSTALL_HEADER_SETS)
+  if(NOT CMAKE_VERSION VERSION_LESS 3.23)
+    get_target_property(INTERFACE_HEADER_SETS ${LIB} INTERFACE_HEADER_SETS)
+    if(INTERFACE_HEADER_SETS)
+      boost_message(DEBUG "boost_install_target: '${__TARGET}' has INTERFACE_HEADER_SETS=${INTERFACE_HEADER_SETS}")
+      set(__INSTALL_HEADER_SETS FILE_SET ${INTERFACE_HEADER_SETS} DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/boost-${__VERSION}")
+    endif()
+  endif()
+
   install(TARGETS ${LIB} EXPORT ${LIB}-targets
     # explicit destination specification required for 3.13, 3.14 no longer needs it
     RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
@@ -344,8 +363,18 @@
     ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
     PRIVATE_HEADER DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}"
     PUBLIC_HEADER DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}"
+    # explicit needed if used starting with cmake v3.28
+    # XXX FILE_SET CXX_MODULES DESTINATION "${CMAKE_INSTALL_DATADIR}"
+    ${__INSTALL_CXX_MODULES}
+    # Any module files from C++ modules from PUBLIC sources in a file set of type CXX_MODULES will be installed to the given DESTINATION.
+    # FIXME: why does this silently not work here? CK
+    # XXX CXX_MODULES_BMI DESTINATION ${CONFIG_INSTALL_DIR}/bmi-${CMAKE_CXX_COMPILER_ID}_$<CONFIG>
+    # explicit needed if used starting with cmake v3.23
+    # XXX FILE_SET HEADERS
+    ${__INSTALL_HEADER_SETS}
   )
 
+  # TODO(CK): what is this for?
   export(TARGETS ${LIB} NAMESPACE Boost:: FILE export/${LIB}-targets.cmake)
 
   if(MSVC)
@@ -362,7 +391,9 @@
     __boost_install_update_sources(${LIB} ${__EXTRA_DIRECTORY} ${__EXTRA_INSTALL_DIRECTORY})
   endif()
 
-  install(EXPORT ${LIB}-targets DESTINATION "${CONFIG_INSTALL_DIR}" NAMESPACE Boost:: FILE ${LIB}-targets.cmake)
+  install(EXPORT ${LIB}-targets DESTINATION "${CONFIG_INSTALL_DIR}" NAMESPACE Boost:: FILE ${LIB}-targets.cmake
+    # XXX CXX_MODULES_DIRECTORY .
+  )
 
   set_target_properties(${LIB} PROPERTIES _boost_is_installed ON)
 
@@ -589,6 +620,7 @@
 
   endif()
 
+  set(extrainstalldir)
   if(__EXTRA_DIRECTORY AND NOT BOOST_SKIP_INSTALL_RULES AND NOT CMAKE_SKIP_INSTALL_RULES)
 
     # Extract the library name from the path, one component up from the extra directory
--- tools/cmake/include/BoostRoot.cmake.old	2026-08-20 18:44:31.506539647 +0200
+++ tools/cmake/include/BoostRoot.cmake	2026-08-17 16:46:20.302542033 +0200
@@ -123,7 +123,9 @@
     set(BOOST_SKIP_INSTALL_RULES ON)
   endif()
 
-  set(BUILD_TESTING OFF)
+  if(NOT BOOST_BUILD_TESTING)
+    set(BUILD_TESTING OFF)
+  endif()
 
 endif()
 
@@ -303,6 +305,9 @@
 
 endmacro()
 
+# C++20 modules
+option(BOOST_USE_MODULES "Build Boost as a collection of C++20 modules (unstable)" OFF)
+
 #
 
 file(GLOB __boost_libraries RELATIVE "${BOOST_SUPERPROJECT_SOURCE_DIR}/libs" "${BOOST_SUPERPROJECT_SOURCE_DIR}/libs/*/CMakeLists.txt" "${BOOST_SUPERPROJECT_SOURCE_DIR}/libs/numeric/*/CMakeLists.txt")
--- libs/serialization/CMakeLists.txt.old	2026-08-19 11:08:14.000000000 +0200
+++ libs/serialization/CMakeLists.txt	2026-08-17 16:46:20.302748163 +0200
@@ -6,6 +6,8 @@
 
 project(boost_serialization VERSION "${BOOST_SUPERPROJECT_VERSION}" LANGUAGES CXX)
 
+set(CMAKE_UNITY_BUILD OFF)
+
 add_library(boost_serialization
   src/archive_exception.cpp
   src/basic_archive.cpp
--- libs/any/CMakeLists.txt.old	2026-08-18 19:27:38.000000000 +0200
+++ libs/any/CMakeLists.txt	2026-08-21 13:15:12.036761186 +0200
@@ -2,25 +2,25 @@
 # Distributed under the Boost Software License, Version 1.0.
 # See accompanying file LICENSE_1_0.txt or copy at https://www.boost.org/LICENSE_1_0.txt
 
-cmake_minimum_required( VERSION 3.8...3.31 )
+cmake_minimum_required(VERSION 3.8...4.4)
 project( boost_any VERSION "${BOOST_SUPERPROJECT_VERSION}" LANGUAGES CXX )
 
-if (BOOST_USE_MODULES)
-    add_library(boost_any)
+if(BOOST_USE_MODULES)
+    add_library(boost_any STATIC)
     target_sources(boost_any PUBLIC
-        FILE_SET modules_public
-            TYPE CXX_MODULES
+        FILE_SET CXX_MODULES
+            BASE_DIRS modules
             FILES modules/boost_any.cppm
     )
 
     target_compile_features(boost_any PUBLIC cxx_std_20)
     target_compile_definitions(boost_any PUBLIC BOOST_USE_MODULES)
-    set(__boost_cxx_standard ${CMAKE_CXX_STANDARD})
-    if (NOT __boost_cxx_standard)
-        set(__boost_cxx_standard 20)
+    if(NOT CMAKE_CXX_STANDARD)
+        set(CMAKE_CXX_STANDARD 20)
     endif()
-    if (CMAKE_CXX_COMPILER_IMPORT_STD AND ${__boost_cxx_standard} IN_LIST CMAKE_CXX_COMPILER_IMPORT_STD)
-        target_compile_definitions(boost_any PRIVATE BOOST_ANY_USE_STD_MODULE)
+    if(CMAKE_CXX_MODULE_STD AND CMAKE_CXX_STANDARD IN_LIST CMAKE_CXX_COMPILER_IMPORT_STD)
+        target_compile_definitions(boost_any PUBLIC BOOST_ANY_USE_STD_MODULE)
+        target_compile_features(boost_any PUBLIC cxx_std_23)
         message(STATUS "Using `import std;`")
     else()
         message(STATUS "`import std;` is not available")
--- libs/conversion/CMakeLists.txt.old	2026-08-18 18:13:14.105023000 +0200
+++ libs/conversion/CMakeLists.txt	2026-08-21 13:09:29.125413790 +0200
@@ -4,23 +4,23 @@
 # Distributed under the Boost Software License, Version 1.0.
 # See accompanying file LICENSE_1_0.txt or copy at https://www.boost.org/LICENSE_1_0.txt
 
-cmake_minimum_required(VERSION 3.5...4.20)
+cmake_minimum_required(VERSION 3.5...4.4)
+
 project(boost_conversion VERSION "${BOOST_SUPERPROJECT_VERSION}" LANGUAGES CXX)
 
-if (BOOST_USE_MODULES)
-  add_library(boost_conversion)
+if(BOOST_USE_MODULES)
+  add_library(boost_conversion STATIC)
   target_sources(boost_conversion PUBLIC
-    FILE_SET modules_public
-    TYPE CXX_MODULES
-    BASE_DIRS modules
-    FILES ${CMAKE_CURRENT_LIST_DIR}/modules/boost_conversion.cppm
+    FILE_SET CXX_MODULES
+      BASE_DIRS modules
+      FILES modules/boost_conversion.cppm
   )
 
   target_compile_definitions(boost_conversion PUBLIC BOOST_USE_MODULES)
   get_property(__standard TARGET boost_conversion PROPERTY CXX_STANDARD)
-  if (__standard IN_LIST CMAKE_CXX_COMPILER_IMPORT_STD AND CMAKE_CXX_MODULE_STD)
+  if(__standard IN_LIST CMAKE_CXX_COMPILER_IMPORT_STD AND CMAKE_CXX_MODULE_STD)
     target_compile_features(boost_conversion PUBLIC cxx_std_23)
-    target_compile_definitions(boost_conversion PRIVATE BOOST_CONVERSION_USE_STD_MODULE)
+    target_compile_definitions(boost_conversion PUBLIC BOOST_CONVERSION_USE_STD_MODULE)
     message(STATUS "Using `import std;`")
   else()
     target_compile_features(boost_conversion PUBLIC cxx_std_20)
--- libs/dll/CMakeLists.txt.old	2026-08-18 17:46:16.508869000 +0200
+++ libs/dll/CMakeLists.txt	2026-08-18 17:33:54.245209309 +0200
@@ -5,7 +5,7 @@
 # Distributed under the Boost Software License, Version 1.0.
 # https://www.boost.org/LICENSE_1_0.txt
 
-cmake_minimum_required(VERSION 3.5...4.20)
+cmake_minimum_required(VERSION 3.5...4.4)
 
 project(boost_dll VERSION "${BOOST_SUPERPROJECT_VERSION}" LANGUAGES CXX)
 
@@ -51,14 +51,14 @@
 
 # Default library
 
-if (BOOST_USE_MODULES)
+if(BOOST_USE_MODULES)
   add_library(boost_dll)
   target_compile_definitions(boost_dll PUBLIC BOOST_USE_MODULES)
 
   get_property(__standard TARGET boost_dll PROPERTY CXX_STANDARD)
-  if (__standard IN_LIST CMAKE_CXX_COMPILER_IMPORT_STD AND CMAKE_CXX_MODULE_STD)
+  if(__standard IN_LIST CMAKE_CXX_COMPILER_IMPORT_STD AND CMAKE_CXX_MODULE_STD)
     target_compile_features(boost_dll PUBLIC cxx_std_23)
-    target_compile_definitions(boost_dll PRIVATE BOOST_DLL_USE_STD_MODULE)
+    target_compile_definitions(boost_dll PUBLIC BOOST_DLL_USE_STD_MODULE)
     message(STATUS "Using `import std;`")
   else()
     target_compile_features(boost_dll PUBLIC cxx_std_20)
@@ -80,7 +80,7 @@
 
 add_library(Boost::dll ALIAS boost_dll)
 target_include_directories(boost_dll ${__scope} include)
-if (BOOST_DLL_USE_STD_FS)
+if(BOOST_DLL_USE_STD_FS)
   target_link_libraries(boost_dll ${__scope} Boost::dll_std_fs)
 else()
   target_link_libraries(boost_dll ${__scope} Boost::dll_boost_fs)
--- libs/lexical_cast/CMakeLists.txt.old	2026-08-17 21:14:44.138016000 +0200
+++ libs/lexical_cast/CMakeLists.txt	2026-08-21 13:09:18.668146913 +0200
@@ -3,22 +3,26 @@
 # Distributed under the Boost Software License, Version 1.0.
 # https://www.boost.org/LICENSE_1_0.txt
 
-cmake_minimum_required(VERSION 3.5...4.0)
+cmake_minimum_required(VERSION 3.5...4.4)
 
 project(boost_lexical_cast VERSION "${BOOST_SUPERPROJECT_VERSION}" LANGUAGES CXX)
 
-
-if (BOOST_USE_MODULES)
-    add_library(boost_lexical_cast)
+if(BOOST_USE_MODULES)
+    add_library(boost_lexical_cast STATIC)
     target_sources(boost_lexical_cast PUBLIC
-        FILE_SET modules_public TYPE CXX_MODULES FILES
-            ${CMAKE_CURRENT_LIST_DIR}/modules/boost_lexical_cast.cppm
+        FILE_SET CXX_MODULES
+            BASE_DIRS modules
+            FILES modules/boost_lexical_cast.cppm
     )
 
     target_compile_features(boost_lexical_cast PUBLIC cxx_std_20)
     target_compile_definitions(boost_lexical_cast PUBLIC BOOST_USE_MODULES)
-    if (CMAKE_CXX_COMPILER_IMPORT_STD)
-        target_compile_definitions(boost_lexical_cast PRIVATE BOOST_LEXICAL_CAST_USE_STD_MODULE)
+    if(NOT CMAKE_CXX_STANDARD)
+        set(CMAKE_CXX_STANDARD 20)
+    endif()
+    if (CMAKE_CXX_MODULE_STD AND CMAKE_CXX_STANDARD IN_LIST CMAKE_CXX_COMPILER_IMPORT_STD)
+        target_compile_definitions(boost_lexical_cast PUBLIC BOOST_LEXICAL_CAST_USE_STD_MODULE)
+        target_compile_features(boost_lexical_cast PUBLIC cxx_std_23)
         message(STATUS "Using `import std;`")
     else()
         message(STATUS "`import std;` is not available")
--- libs/pfr/CMakeLists.txt.old	2026-08-18 19:29:18.000000000 +0200
+++ libs/pfr/CMakeLists.txt	2026-08-21 13:10:00.727689106 +0200
@@ -4,20 +4,26 @@
 # Distributed under the Boost Software License, Version 1.0.
 # https://www.boost.org/LICENSE_1_0.txt
 
-cmake_minimum_required(VERSION 3.5...3.31)
+cmake_minimum_required(VERSION 3.28...4.4)
 
 project(boost_pfr VERSION "${BOOST_SUPERPROJECT_VERSION}" LANGUAGES CXX)
 
-if (BOOST_USE_MODULES)
-    add_library(boost_pfr)
+if(BOOST_USE_MODULES)
+    add_library(boost_pfr STATIC)
     target_sources(boost_pfr PUBLIC
-        FILE_SET modules_public TYPE CXX_MODULES FILES modules/boost_pfr.cppm
+        FILE_SET CXX_MODULES
+            BASE_DIRS modules
+            FILES modules/boost_pfr.cppm
     )
 
     target_compile_features(boost_pfr PUBLIC cxx_std_20)
     target_compile_definitions(boost_pfr PUBLIC BOOST_USE_MODULES)
-    if (CMAKE_CXX_COMPILER_IMPORT_STD)
-        target_compile_definitions(boost_pfr PRIVATE BOOST_PFR_USE_STD_MODULE)
+    if(NOT CMAKE_CXX_STANDARD)
+        set(CMAKE_CXX_STANDARD 20)
+    endif()
+    if(CMAKE_CXX_MODULE_STD AND CMAKE_CXX_STANDARD IN_LIST CMAKE_CXX_COMPILER_IMPORT_STD)
+        target_compile_definitions(boost_pfr PUBLIC BOOST_PFR_USE_STD_MODULE)
+        target_compile_features(boost_pfr PUBLIC cxx_std_23)
         message(STATUS "Using `import std;`")
     else()
         message(STATUS "`import std;` is not available")
@@ -31,11 +37,11 @@
 add_library(Boost::pfr ALIAS boost_pfr)
 
 enable_testing()
-if (BUILD_TESTING AND EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/test/CMakeLists.txt")
+if(BUILD_TESTING AND EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/test/CMakeLists.txt")
     add_subdirectory(test)
 endif()
 
-if (BOOST_USE_MODULES AND BUILD_TESTING)
+if(BOOST_USE_MODULES AND BUILD_TESTING)
     add_executable(boost_pfr_module_usage modules/usage_sample.cpp)
     target_link_libraries(boost_pfr_module_usage PRIVATE Boost::pfr)
     add_test(NAME boost_pfr_module_usage COMMAND boost_pfr_module_usage)
--- libs/type_index/CMakeLists.txt.old	2026-08-17 21:36:16.377197000 +0200
+++ libs/type_index/CMakeLists.txt	2026-08-21 13:10:17.933107868 +0200
@@ -4,24 +4,26 @@
 # Distributed under the Boost Software License, Version 1.0.
 # https://www.boost.org/LICENSE_1_0.txt
 
-cmake_minimum_required(VERSION 3.5...4.20)
+cmake_minimum_required(VERSION 3.5...4.4)
 
 project(boost_type_index VERSION "${BOOST_SUPERPROJECT_VERSION}" LANGUAGES CXX)
 
-if (BOOST_USE_MODULES)
-    add_library(boost_type_index)
-    target_sources(boost_type_index
-      PUBLIC
+if(BOOST_USE_MODULES)
+    add_library(boost_type_index STATIC)
+    target_sources(boost_type_index PUBLIC
         FILE_SET CXX_MODULES
-        BASE_DIRS "${CMAKE_CURRENT_SOURCE_DIR}"
-        FILES "${CMAKE_CURRENT_SOURCE_DIR}/modules/boost_type_index.cppm"
+            BASE_DIRS modules
+            FILES "modules/boost_type_index.cppm"
     )
 
     target_compile_features(boost_type_index PUBLIC cxx_std_20)
     target_compile_definitions(boost_type_index PUBLIC BOOST_USE_MODULES)
-    if ((CMAKE_CXX_STANDARD IN_LIST CMAKE_CXX_COMPILER_IMPORT_STD) AND CMAKE_CXX_MODULE_STD)
-        target_compile_features(boost_type_index PRIVATE cxx_std_23)
-        target_compile_definitions(boost_type_index PRIVATE BOOST_TYPE_INDEX_USE_STD_MODULE)
+    if(NOT CMAKE_CXX_STANDARD)
+        set(CMAKE_CXX_STANDARD 20)
+    endif()
+    if(CMAKE_CXX_MODULE_STD AND CMAKE_CXX_STANDARD IN_LIST CMAKE_CXX_COMPILER_IMPORT_STD)
+        target_compile_features(boost_type_index PUBLIC cxx_std_23)
+        target_compile_definitions(boost_type_index PUBLIC BOOST_TYPE_INDEX_USE_STD_MODULE)
         message(STATUS "Using `import std;`")
     else()
         message(STATUS "`import std;` is not available")
